File: java/words/src/main/java/com/example/words/Numerical.java
package com.example.words;

public class Numerical {
	public static int fibonacci(int n)
	{
		if (n < 0)
		{
			return -1;
		}
		if (n <= 2)
		{
			return n;
		}
		int acc1 = 1;
		int acc2 = 0;
		int aux;
		for (int i = 3; i <= n; i++)
		{
			aux = acc1 + acc2;
			acc2 = acc1;
			acc1 = aux;
		}
		return acc1 + acc2;
	}

	public static int factorial(int n)
	{
		if (n < 0)
		{
			return -1;
		}
		if (n == 0 || n == 1)
		{
			return 1;
		}
		int acc = 1;
		for (int i = 2; i <= n; i++)
		{
			acc *= i;
		}
		return acc;
	}

	public static int gcd(int a, int b)
	{
		int aux;
		while (b != 0)
		{
			aux = a % b;
			a = b;
			b = aux;
		}
		return a;
	}

	public static int lcm(int a, int b)
	{
		return (a / gcd(a, b)) * b;
	}
}

File: _experimental/_swift/words/Sources/numerical/numerical.swift
public struct Numerical {
	public static func fibonacci(n: Int) -> Int {
		if n < 0 { return -1 }
		if n <= 1 { return n }
		var acc1 = 1
		var acc2 = 0
		var aux = 0
		for _ in 2..<n {
			aux = acc1
			acc1 = acc1 + acc2
			acc2 = aux
		}
		return acc1 + acc2
	}

	public static func factorial(n: Int) -> Int {
		if n < 0 { return -1 }
		var acc = 1
		for i in 2..<(n + 1) {
			acc = acc * i
		}
		return acc
	}

	public static func gcd(a: Int, b: Int) -> Int {
		if b == 0 { return a }
		return gcd(a: b, b: a % b)
	}

	public static func lcm(a: Int, b: Int) -> Int {
		return (a / gcd(a: a, b: b)) * b
	}
}